src/Components/WebAssembly/WebAssembly/src/HotReload/WebAssemblyHotReload.cs

```diff
@@ -43,7 +43,7 @@ internal static async Task InitializeAsync()
     /// For framework use only.
     /// </summary>
     [JSInvokable(nameof(ApplyHotReloadDelta))]
-    public static void ApplyHotReloadDelta(string moduleIdString, byte[] metadataDelta, byte[] ilDelta, byte[] pdbBytes)
+    public static void ApplyHotReloadDelta(string moduleIdString, byte[] metadataDelta, byte[] ilDelta, byte[] pdbBytes, int[] updatedTypes)
```

Comment:
I'm not sure how the public API policy works here. The JS invoker will likely be able to invoke the method without a type, but there may be a need for a separate method that matches previous versions of VS (which injects the browserlink payload).

Reply:
Rather than having two separate methods, could we allow the `updatedTypes` parameter to be an `int[]?`? It seems like this logic should still work even if the updated types are not provided by the scripts injected by the SDK. And we could make updated `Blazor._internal.applyHotReload` implementation pass `updatedTypes ?? null` in the call to `dispatcher.invokeDotNetStaticMethod`.

This is technically a breaking API change, but that should be fine in this case because the API is meant for framework use only (customers should not be relying on it).

If you are able to open the solution in VS, there should be a codefix to update the `PublicAPI.*.txt` files to reflect the public API changes. If you don't have this set up, I'd be happy to update those files and add the changes to this PR.

Reply:
Thanks for the review! I was indeed able to run the code fix, it added an entry in the unshipped file.